ikarus/finiteelements/mechanics/enhancedassumedstrains.hh

```diff
@@ -158,43 +162,48 @@ forward the
+  template <typename RT, bool voigt = true>
+  auto calculateAt(const FERequirementType& req, const Dune::FieldVector<double, Traits::mydim>& local) const
+      -> ResultTypeType<RT, voigt> {
     using namespace Dune::Indices;
     using namespace Dune::DerivativeDirections;
-    using namespace Dune;
 
-    static_assert(resType == ResultType::linearStress, "The requested result type is NOT implemented.");
-
-    auto resultVector = DisplacementBasedElement::template calculateAt<resType>(req, local);
-
-    if (isDisplacementBased())
-      return resultVector;
-
-    const auto C              = DisplacementBasedElement::materialTangentFunction(req);
-    const auto& numberOfNodes = DisplacementBasedElement::numberOfNodes();
-    auto uFunction            = DisplacementBasedElement::displacementFunction(req);
-    const auto disp           = Dune::viewAsFlatEigenVector(uFunction.coefficientsRef());
-
-    std::visit(
-        [&]<typename EAST>(const EAST& easFunction) {
-          if constexpr (not std::is_same_v<std::monostate, EAST>) {
-            constexpr int enhancedStrainSize = EAST::enhancedStrainSize;
-            Eigen::Matrix<double, enhancedStrainSize, enhancedStrainSize> D;
-            calculateDAndLMatrix(easFunction, req, D, L_);
-            const auto alpha = (-D.inverse() * L_ * disp).eval();
-            const auto M     = easFunction.calcM(local);
-            const auto CEval = C(local);
-            auto easStress   = (CEval * M * alpha).eval();
-
-            resultVector.resize(3, 1);
-            resultVector = resultVector + easStress;
-          }
-        },
-        easVariant_);
-    return resultVector;
+    if constexpr (std::is_same_v<RT, ResultType::linearStress>) {
+      if (isDisplacementBased())
+        return DisplacementBasedElement::template calculateAt<RT, voigt>(req, local);
+
+      const auto C              = DisplacementBasedElement::materialTangentFunction(req);
+      const auto& numberOfNodes = DisplacementBasedElement::numberOfNodes();
+      auto uFunction            = DisplacementBasedElement::displacementFunction(req);
+      const auto disp           = Dune::viewAsFlatEigenVector(uFunction.coefficientsRef());
+
+      ResultTypeType<RT, voigt> result;
+      std::visit(
+          [&]<typename EAST>(const EAST& easFunction) {
+            if constexpr (not std::is_same_v<std::monostate, EAST>) {
+              constexpr int enhancedStrainSize = EAST::enhancedStrainSize;
+              Eigen::Matrix<double, enhancedStrainSize, enhancedStrainSize> D;
+              calculateDAndLMatrix(easFunction, req, D, L_);
+              const auto alpha = (-D.inverse() * L_ * disp).eval();
+              const auto M     = easFunction.calcM(local);
+              const auto CEval = C(local);
+              auto easStress   = (CEval * M * alpha).eval();
+
+              if constexpr (voigt)
```

Comment:
Im not so happy by doing this decition in each element. Maybe we invert the mechanism here, that the Element provides the results in the most convenient way and at the outside the user has to transform it if needed. WE can discuss tomorrow maybe.